### ISSUE TITLE
Planning / First steps

### ISSUE THREAD

**Issue by kescobo** [carpentries-incubator/julia-novice#2]
I'm not 100% sure what the firs steps for getting going on this, or who should take the initiative, so I figured I could at least open an issue :). 

There was talk over on julia discourse about doing both a software and data carpentry lesson on Julia, any reason not to just develop them both in the same repo?

**Comment by giordano** [comment]
I wonder if @tobyhodges can comment on the last point :slightly_smiling_face:

**Comment by tobyhodges** [comment]
Each Carpentries lesson is hosted via GitHub Pages, with Software, Library, and Data Carpentry lessons using the same lesson template. As only one branch can be used for Pages on any given repository, it would be difficult to develop two separate lessons in the same repo.

**Comment by kescobo** [comment]
Ahh, thanks for the explanation @tobyhodges, that makes sense. Is the best way forward then to open a new issue in the proposals repo for a Data Carpentry with Julia?

Perhaps rename this repo to sc-julia-novice, and I'll propose dc-julia-novice?

**Comment by tobyhodges** [comment]
Yes, best to open a new proposal to the Incubator.

As the intention is for this lesson to eventually join the Software Carpentry lesson program, the name of this repo was chosen to be (somewhat) in line with the naming conventions of other Software Carpentry programming lessons (e.g. `python-novice-inflammation` and `python-novice-gapminder`. We couldn't choose a last part to the name because we didn't know what data would be used for examples in the lesson (this is the `inflammation` and `gapminder` parts of those other lesson slugs).

However, Data Carpentry lessons are typically focused on a particular type of data or domain: the language/tool is taught as a means to manage/organise/filter/clean/analyse that data. This is in contrast to Software Carpentry, where programming lessons are focussed more on teaching good coding practices to produce reproducible/re-usable programs. For example, Data Carpentry has R lessons for ecology data, genomics data, social science data, geospatial data, etc. Those lessons teach similar things, but the lessons are always domain-focused.

So, if you would like to create a Data Carpentry lesson that uses Julia (which I think would be a great idea!), I would recommend thinking in advance about what the narrative will be for that lesson. What kind of data set would the lesson use? What kind of analysis would learners perform during the lesson? What sorts of figures would they create by the end? What kind of steps would they need to take to get there? What features of the Julia language are most useful for a member of the target audience to learn to be ready to work with their data?

I hope that made sense? Happy to provide more info/expand further as required. But probably not today - it's almost dinner time here 😄

**Comment by kescobo** [comment]
Definitely, thanks! I'll spend some more time with some of the existing lessons and then make a proposal.

**Comment by tobyhodges** [comment]
I should stress that you do not need to have answers to all (any) of my questions above before proposing the lesson to the Incubator. You can open up a repository there first and then figure out the narrative for the lesson afterwards if you prefer. But I wanted to make the distinction clear between Software and Data Carpentry lessons to help you decide whether you want to work on two separate Julia lessons or combine efforts on this single repository.
